core/src/main/kotlin/io/fritz2/binding/substores.kt

```diff
@@ -0,0 +1,39 @@
+package io.fritz2.binding
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+
+interface Lens<P,T> {
+    fun get(parent: P): T
+    fun set(parent: P, value: T): P
+    fun map(parent: P, mapper: (T) -> T): P = set(parent, mapper(get(parent)))
+
+
+    operator fun <X> plus(other: Lens<T,X>): Lens<P,X> = object : Lens<P,X> {
+        override fun get(parent: P): X = other.get(this@Lens.get(parent))
+        override fun set(parent: P, value: X): P = this@Lens.set(parent, other.set(this@Lens.get(parent), value))
+    }
+}
```

Comment:
Maybe add some generic id generation here?

Reply:
definitely! I think this should be done in code-generation by just adding an ´´´val id = "${parent.id}.<generated id>"```